src/aks-preview/azext_aks_preview/aks_app_commands/aksappcommands.py

```diff
@@ -0,0 +1,150 @@
+# pylint: disable=too-many-lines
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import sre_parse
+from typing import Dict, List
+import shutil
+import subprocess
+import requests
+import os
+import platform
+from pathlib import Path
+from knack.prompting import prompt_y_n
+
+
+def aks_draft_app_init(deployment_path: str,
+                       app_name: str,
+                       language: str,
+                       create_config: str,
+                       dockerfile_only: str,
+                       deployment_only: str) -> None:
+    file_path = _binary_pre_check()
+    if not file_path:
+        raise ValueError('Binary was NOT executed successfully')
+
+    arguments = _build_arguments(app_name, language, create_config, dockerfile_only, deployment_only)
+    run_successful = _run(file_path, deployment_path, arguments)
+    if run_successful:
+        _cmd_finish()
+
+
+# If setup is valid this method returns the correct binary path to execute
+def _binary_pre_check() -> str:
+    print('The DraftV2 setup is in progress...')
+    draftv2_binary_path = _get_existing_path()
+
+    if draftv2_binary_path:  # use found binary
+        return draftv2_binary_path
+    else:  # prompt user to download binary
+        return _download_binary()
+
+
+# Returns path to existing draftv2 binary and None otherwise
+def _get_existing_path() -> str:
+    print('Checking if DraftV2 binary exists locally...')
+
+    operating_system = platform.system().lower()
+    # Filename depends on the operating system
+    filename = 'draftv2-' + operating_system + '-amd64'
```

Comment:
💡 This is `amd64` arch based so is this only going to work for some OS and not all?